packages/compiler/core/checker.ts

```diff
@@ -2272,6 +2321,86 @@ export function createChecker(program: Program): Checker {
     properties.set(newProp.name, newProp);
   }
 
+  function bindAllMembers(node: Node) {
+    if (node.symbol) {
+      bindMembers(node, node.symbol);
+    }
+    visitChildren(node, (child) => {
+      bindAllMembers(child);
+    });
+  }
+
+  function bindMembers(node: Node, containerSym: Sym) {
+    let containerMembers: Mutable<SymbolTable>;
+
+    switch (node.kind) {
+      case SyntaxKind.ModelStatement:
+        if (node.extends && node.extends.kind === SyntaxKind.TypeReference) {
+          resolveAndCopyMembers(node.extends);
+        }
+        if (node.is && node.is.kind === SyntaxKind.TypeReference) {
+          resolveAndCopyMembers(node.is);
+        }
+        for (const prop of node.properties) {
+          if (prop.kind === SyntaxKind.ModelSpreadProperty) {
+            resolveAndCopyMembers(prop.target);
+          } else {
+            const name = prop.id.kind === SyntaxKind.Identifier ? prop.id.sv : prop.id.value;
+            bindMember(name, prop, SymbolFlags.ModelProperty);
+          }
+        }
+        break;
+      case SyntaxKind.EnumStatement:
+        for (const member of node.members.values()) {
+          if (member.kind === SyntaxKind.EnumSpreadMember) {
+            resolveAndCopyMembers(member.target);
+          } else {
+            const name = member.id.kind === SyntaxKind.Identifier ? member.id.sv : member.id.value;
+            bindMember(name, member, SymbolFlags.EnumMember);
+          }
+        }
+        break;
+      case SyntaxKind.InterfaceStatement:
+        for (const member of node.operations.values()) {
+          bindMember(member.id.sv, member, SymbolFlags.InterfaceMember);
+        }
+        break;
+      case SyntaxKind.UnionStatement:
+        for (const variant of node.options.values()) {
+          const name = variant.id.kind === SyntaxKind.Identifier ? variant.id.sv : variant.id.value;
+          bindMember(name, variant, SymbolFlags.UnionVariant);
+        }
+        break;
+    }
+
+    // TODO: PR check if we prefer this alternative to dup
+    // program.reportDuplicateSymbols(getOrCreateAugmentedSymbolTable(containerSym.members!));
```

Comment:
Think this is a potential alternative. Problem It still won't work if there is late bound members(Spread template) so not sure its worth

Reply:
Can you say more about this, I don't understand the alternatives.

Reply:
Right now the duplicate member is done at the checkModelProperties (or checkEnumMembers for enum etc)

At this point now we have a better picture of all the members in a model, enum ,etc. The only time where we don't know the full member list is:

```
model Foo<T> {
  ...T
}
```

here we can't know the members until we have instantiated it.

I was kinda wondering if we should reuse the symbol duplicate logic that is already existing but the problem is that case above where we'd still need to double check. So I guess I don't think it is super worth it.

Reply:
I see. If it it can't replace the existing check, then yes, I agree it isn't worth it.